src/derivative_operators/vector_algebraic_operations.jl

```diff
@@ -35,10 +35,8 @@ function cross_product!(u::AbstractArray{T1,4},A::AbstractArray{T2,4},B::Abstrac
     
     (size(A) === size(B) && size(A) === size(u))|| throw(ArgumentError("Vectors must have the same shape"))
     s = size(u)
-    for p in 1:s[1],q in 1:s[2],r in 1:s[3]
-        for i in 1:3
-            u[p,q,r,i] = A[p,q,r,i%3 + 1]*B[p,q,r,(i+1)%3 + 1] - A[p,q,r,(i+1)%3 + 1]*B[p,q,r,i%3 + 1]
-        end
+    for i in 1:3, r in 1:s[3], q in 1:s[2], p in 1:s[1]
+        u[p,q,r,i] = A[p,q,r,i%3 + 1]*B[p,q,r,(i+1)%3 + 1] - A[p,q,r,(i+1)%3 + 1]*B[p,q,r,i%3 + 1]
```

Comment:
I think it might be more efficient to hoist the `i%3 + 1` out of the inner loop? Mod operations are expensive. Actually, if you can write that without a mod or an if statement that would be best.